samples/sample_common/include/brc_routines.h

```diff
@@ -401,6 +483,61 @@ namespace HEVCExtBRC
         return MFX_ERR_NONE;
     }
 }
+namespace HEVCExtBRCUserCQP
+{
+    inline mfxStatus Init(mfxHDL pthis, mfxVideoParam* par)
+    {
+        MFX_CHECK_NULL_PTR1(pthis);
+        return ((ExtBRCUserCQP*)pthis)->Init(par);
+    }
+    inline mfxStatus Reset(mfxHDL pthis, mfxVideoParam* par)
+    {
+        MFX_CHECK_NULL_PTR1(pthis);
+        return ((ExtBRCUserCQP*)pthis)->Reset(par);
+    }
+    inline mfxStatus Close(mfxHDL pthis)
+    {
+        MFX_CHECK_NULL_PTR1(pthis);
+        return ((ExtBRCUserCQP*)pthis)->Close();
+    }
+
+    inline mfxStatus GetFrameCtrl(mfxHDL pthis, mfxBRCFrameParam* par, mfxBRCFrameCtrl* ctrl)
+    {
+        MFX_CHECK_NULL_PTR1(pthis);
+        return ((ExtBRCUserCQP*)pthis)->GetFrameCtrl(par, ctrl);
+    }
+    inline mfxStatus Update(mfxHDL pthis, mfxBRCFrameParam* par, mfxBRCFrameCtrl* ctrl, mfxBRCFrameStatus* status)
+    {
+        MFX_CHECK_NULL_PTR1(pthis);
+        return ((ExtBRCUserCQP*)pthis)->Update(par, ctrl, status);
+    }
+
+    inline mfxStatus Create(mfxExtBRC & m_BRC)
+    {
+        MFX_CHECK(m_BRC.pthis == NULL, MFX_ERR_UNDEFINED_BEHAVIOR);
+        m_BRC.pthis = new ExtBRCUserCQP;
+        m_BRC.Init = Init;
+        m_BRC.Reset = Reset;
+        m_BRC.Close = Close;
+        m_BRC.GetFrameCtrl = GetFrameCtrl;
+        m_BRC.Update = Update;
+        return MFX_ERR_NONE;
+    }
+    inline mfxStatus Destroy(mfxExtBRC & m_BRC)
+    {
+        if (m_BRC.pthis != NULL)
```

Comment:
deleting null pointer is safe and well defined procedure which simply does nothing